app/controllers/admin/settings/project_custom_fields_controller.rb

```diff
@@ -68,13 +71,18 @@ def edit; end
     def project_mappings; end
 
     def unlink
-      project = Project.find(permitted_params.project_custom_field_project_mapping[:project_id])
-      project_custom_field_mapping = @custom_field.project_custom_field_project_mappings.find_by(project:)
       delete_service = ProjectCustomFieldProjectMappings::DeleteService
-                         .new(user: current_user, model: project_custom_field_mapping)
+                         .new(user: current_user, model: @project_custom_field_mapping)
                          .call
 
-      delete_service.on_success { render_unlink_response(project:) }
+      delete_service.on_success { render_unlink_response(project: @project) }
+
+      delete_service.on_failure do
+        update_flash_message_via_turbo_stream(
+          message: join_flash_messages(delete_service.errors.full_messages),
+          full: true, dismiss_scheme: :hide, scheme: :danger
+        )
+      end
```

Comment:
🟢 I think this combination is prone to errors if a developer tries to remove one branch. They only make sense in tandem. That's why I introduced `match` for this case. It enforces the developer to take care.

Reply:
Good call 👍🏾

Reply:
I got some errors here, will reach out to you later to understand but will merge for now.